### PR TITLE
Refactor request validation, and make error codes more consistent

### LINKED ISSUE #98: review the payload validation
We're validating the body payload twice. For instance:
```js
router.post(
  '/signup/email-password',
  createValidator().body(signUpEmailPasswordSchema),
  aw(signUpEmailPasswordHandler)
);
```
1. `createValidator().body(signUpEmailPasswordSchema) `
2. in `signUpEmailPasswordHandler` with `isValidEmail`, `isPasswordValid`, `isValidRedirectTo`, and  `isRolesValid`

We should streamline validation with Joi

### PR DESCRIPTION
**Pull request by plmercereau**
- consolidate error messages
- redirect errors to the frontend client when redirectTo is an option
- use Joi to validate all the requests before handling the business logic
- move the validation logic in each route file for more readability
- remove unused packages `express-boom` and `express-joi-validation`
- use `http-status-codes` to increase code readability
- restructure helpers and utils
- remove unused code and comments
- prepare the fixing of allowed locales

### Checks
- [x] Jest tests
- [x] Local OAuth tests
- [x] Local Docker build + test

### Post-merge todo list
- [ ] sync error codes/messages in the SDK
- [ ] update Swagger / OpenApi specs in the documentation (generate with a custom CLIENT_URL)
- [ ] create changelog
- [ ] consider using changesets instead of release-it

Closes #98 #46 #84 #116

### CODE REVIEW

File: src/errors.ts [hunk 1]
```diff
@@ -15,10 +18,141 @@ export async function errors(
...
+  status: StatusCodes.BAD_REQUEST,
+  error: 'request-validation-error',
+  message: 'The request payload is incorrect',
+};
+
+const asErrors = <T>(et: {
+  [K in keyof T]: Pick<ErrorPayload, 'status' | 'message'>;
+}) => et;
+
+const ERRORS = asErrors({
```
Comment (elitan) [inline src/errors.ts:46]: How about we use constants to set the keys?

Like:

```
export const ROUTE_NOT_FOUND = 'route-not-found';
```

Then we can use `ROUTE_NOT_FOUND` both in the key here and wherever it's used using the `sendError` function?

File: src/routes/signin/providers/utils.ts [hunk 2]
```diff
@@ -103,8 +107,9 @@ const manageProviderStrategy =
     }
 
     if (email) {
-      // check email
-      if (!(await isValidEmail({ email }))) {
+      try {
+        await emailValidator.validateAsync(email);
+      } catch {
         return done(new Error('email is not allowed'));
```
Comment (elitan) [inline src/routes/signin/providers/utils.ts:113]: Ideally, this error should result in a redirect to the client's app to show a nice error message. I'm not sure it's possible given we don't have access to `res` here.

File: src/routes/user/locale.ts [hunk 3]
```diff
@@ -1,46 +0,0 @@
-import { gqlSdk } from '@/utils/gqlSDK';
```
Comment (elitan) [inline src/routes/user/locale.ts:1]: Why was this file removed?

**Review by elitan — CHANGES_REQUESTED** [review]
Lots of optimizations. Nice!

File: src/errors.ts (hunk 1 above)
Comment (elitan) [reply src/errors.ts:46]: nvm, we have types.

File: src/routes/user/locale.ts (hunk 3 above)
Comment (plmercereau) [reply src/routes/user/locale.ts:1]: This handler was not activated anywhere (not linked to an express route)

File: src/routes/signin/providers/utils.ts (hunk 2 above)
Comment (plmercereau) [reply src/routes/signin/providers/utils.ts:113]: Yes, this is something we should improve, I haven't found a simple way to implement this behaviour

**Review by elitan — APPROVED** [review]
